pkg/api/registry/registry.proto

```diff
@@ -22,61 +22,40 @@ message Destination {
 }
 
-message NetworkServiceManager {
-    string name = 1;
-    string url = 2;
-    google.protobuf.Timestamp expiration_time = 3;
-    string state = 4;
+message NetworkServiceLabels {
+    map<string, string> labels = 1;
 }
 
 message NetworkServiceEndpoint {
     string name = 1;
-    string payload = 2;
-    string network_service_name = 3;
-    string network_service_manager_name = 4;
-    map<string, string> labels = 5;
-    string state = 6;
-
+    repeated string network_service_name = 2;
+    map<string, NetworkServiceLabels> network_service_labels = 3;
     // Endpoint callback address, could be any of unix/tcp/callback schema ones.
-    string url = 7;
-}
-
-message FindNetworkServiceRequest {
-    string network_service_name = 1;
-}
-
-message FindNetworkServiceResponse {
-    string payload = 1;
-    NetworkService network_service = 2;
-    map<string, NetworkServiceManager> network_service_managers = 3;
-    repeated NetworkServiceEndpoint network_service_endpoints = 4;
+    string url = 4;
+    // Network service manager URL for the endpoint.
+    string network_service_manger_url = 5;
+    google.protobuf.Timestamp expiration_time = 6;
 }
 
-message NSERegistration {
-    NetworkService network_service = 1;
-    NetworkServiceManager network_service_manager = 2;
-    NetworkServiceEndpoint network_service_endpoint = 3;
-}
-
-message RemoveNSERequest {
-    string network_service_endpoint_name = 1;
+service NetworkServiceEndpointRegistry {
+    rpc Register (NetworkServiceEndpoint) returns (NetworkServiceEndpoint);
+    rpc Monitor (NetworkServiceEndpointQuery) returns (stream NetworkServiceEndpoint);
+    rpc Find (NetworkServiceEndpointQuery) returns (stream NetworkServiceEndpoint);
+    rpc Unregister (NetworkServiceEndpoint) returns (google.protobuf.Empty);
 }
 
 service NetworkServiceRegistry {
-    rpc RegisterNSE (NSERegistration) returns (NSERegistration);
-    rpc BulkRegisterNSE (stream NSERegistration) returns (stream NSERegistration);
-    rpc RemoveNSE (RemoveNSERequest) returns (google.protobuf.Empty);
-}
-
-service NetworkServiceDiscovery {
-    rpc FindNetworkService (FindNetworkServiceRequest) returns (FindNetworkServiceResponse);
+    rpc Register (NetworkService) returns (NetworkService);
+    rpc Monitor (NetworkServiceQuery) returns (stream NetworkService);
+    rpc Find (NetworkServiceQuery) returns (stream NetworkService);
+    rpc Unregister (NetworkService) returns (google.protobuf.Empty);
 }
 
-message NetworkServiceEndpointList {
-    repeated NetworkServiceEndpoint network_service_endpoints = 1;
+message NetworkServiceQuery {
+    NetworkService network_service = 1;
 }
 
-service NsmRegistry {
-    rpc RegisterNSM (NetworkServiceManager) returns (NetworkServiceManager);
-    rpc GetEndpoints (google.protobuf.Empty) returns (NetworkServiceEndpointList);
-}
+message NetworkServiceEndpointQuery {
+    NetworkServiceEndpoint network_service_endpoint = 1;
+    NetworkService network_service = 2;
```

Comment:
I still don't follow why we have both the NetworkServiceEndpoint and NetworkService objects in the NetworkServiceEndpoint query.  Could you explain?

Reply:
Sure, is this case make sense? https://github.com/networkservicemesh/api/pull/41#issuecomment-638912388

Reply:
Fixed https://github.com/networkservicemesh/api/pull/41/commits/3b53da4e4e1a26185f782b92ce1ca044b516c38a

Reply:
Ah... you took my question as direction :)

So my thought is this... that anything NetworkService related here we'd need to communicate can be done with the NetworkServiceEndpoint.NetworkService field

Reply:
@edwarnicke NetworkServiceEndpoint has not field NetworkService. It has only the name of network service

Reply:
Yes... but other than the name... what else would we look up a NetworkServiceEndpoint on :)